linq4j/src/test/java/org/apache/calcite/linq4j/test/JoinPreserveOrderTest.java

```diff
@@ -141,81 +134,162 @@ public JoinPreserveOrderTest(
                     deptOrderColSelectors.get(j),
                     ascendingR,
                     nullsFirstR);
-                data.add(params);
+                data.add(Arguments.of(params[0], params[1]));
               }
             }
           }
         }
       }
     }
-    return data;
+    return data.stream();
   }
 
-  @Test public void testLeftJoinPreservesOrderOfLeftInput() {
+  private void initColumns(
+      FieldCollationDescription<Employee> leftColumn,
+      FieldCollationDescription<Department> rightColumn) {
+    this.leftColumn = leftColumn;
+    this.rightColumn = rightColumn;
+  }
+
+  @ParameterizedTest
+  @MethodSource("data")
+  void testLeftJoinPreservesOrderOfLeftInput(
+      FieldCollationDescription<Employee> leftColumn,
+      FieldCollationDescription<Department> rightColumn) {
+    initColumns(leftColumn, rightColumn);
     testJoin(hashJoin(false, true), AssertOrder.PRESERVED, AssertOrder.IGNORED);
   }
 
-  @Test public void testRightJoinPreservesOrderOfLeftInput() {
+  @ParameterizedTest
+  @MethodSource("data")
+  void testRightJoinPreservesOrderOfLeftInput(
+      FieldCollationDescription<Employee> leftColumn,
+      FieldCollationDescription<Department> rightColumn) {
+    initColumns(leftColumn, rightColumn);
```

Comment:
It becomes a bit too verbose. Is there an option to avoid this duplication all over the place? :(

Reply:
> It becomes a bit too verbose. Is there an option to avoid this duplication all over the place? :(

Well, I am considering this too. At present I am not very familiar with junit5, I will continue to look for it.

Reply:
I've renamed `FieldCollationDescription` to just `Field`, and it fits on a single line now.